hack/make/build-rpm

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+set -e
+
+DEST=$1
+
+# subshell so that we can export PATH without breaking other things
+(
+	source "$(dirname "$BASH_SOURCE")/.integration-daemon-start"
+
+	# TODO consider using frozen images for the dockercore/builder-rpm tags
+		
+	rpmVersion="${VERSION//-/_}"
+	# if we have a "-dev" suffix or have change in Git, let's make this package version more complex so it works better
+	if [[ "$VERSION" == *-dev ]] || [ -n "$(git status --porcelain)" ]; then
+		gitUnix="$(git log -1 --pretty='%at')"
+		gitDate="$(date --date "@$gitUnix" +'%Y%m%d.%H%M%S')"
+		gitCommit="$(git log -1 --pretty='%h')"
+		gitVersion="git${gitDate}.0.${gitCommit}"
+		# gitVersion is now something like 'git20150128.112847.0.17e840a'
+		rpmVersion="${rpmVersion}_${gitVersion}"
```

Comment:
ah right. this is something we do by hand for in %{release} incrementing.

On Fri, May 1, 2015 at 6:12 PM, Tianon Gravi <notifications@github.com>
wrote:

> In hack/make/build-rpm
> <https://github.com/docker/docker/pull/12917#discussion_r29537579>:
>
> > +
> > +# subshell so that we can export PATH without breaking other things
> > +(
> > +	source "$(dirname "$BASH_SOURCE")/.integration-daemon-start"
> > +
> > +	# TODO consider using frozen images for the dockercore/builder-rpm tags
> > +		
> > +	rpmVersion="${VERSION//-/_}"
> > +	# if we have a "-dev" suffix or have change in Git, let's make this package version more complex so it works better
> > +	if [[ "$VERSION" == *-dev ]] || [ -n "$(git status --porcelain)" ]; then
> > +		gitUnix="$(git log -1 --pretty='%at')"
> > +		gitDate="$(date --date "@$gitUnix" +'%Y%m%d.%H%M%S')"
> > +		gitCommit="$(git log -1 --pretty='%h')"
> > +		gitVersion="git${gitDate}.0.${gitCommit}"
> > +		# gitVersion is now something like 'git20150128.112847.0.17e840a'
> > +		rpmVersion="${rpmVersion}_${gitVersion}"
>
> The point of the date is so that `git20150101` gets replaced by
> `git20150202`, since Git commit IDs aren't naturally sortable. [image:
> :smile:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/12917/files#r29537579>.
>